tests/socket_test.c

```diff
@@ -79,12 +79,18 @@ TEST test_sockopt()
    PASS();
 }
 
+int dup_fd = 0;
+
 int TEST_PORT;
 void* tcp_server_main(void* arg)
 {
    static int rc = 0;
    int fd = *(int*)arg;
 
+   if (dup_fd != 0) {
+      fd = fcntl(fd, F_DUPFD_CLOEXEC, 0);
```

Comment:
You are just causing the open fd's sockinfo to be copied to the new fd.
What part of this test verifies that it is there in the new fd?